### ISSUE TITLE
ossec/queue Errors When Updating to 3.8.0 on Debian

### ISSUE THREAD

**Issue by Justangatang** [wazuh/wazuh#2422]
**OS**
Debian 9

**Wazuh version**
3.8.0_1

**Wazuh API version**
3.8.0_1

**Install type**
Manager

**Install method**
Package

**Log sample**
```
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"(1210): Queue '/queue/ossec/queue' not accessible: 'Connection refused'."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"(1210): Queue '/queue/ossec/queue' not accessible: 'Bad file descriptor'."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"(1210): Queue '/queue/ossec/queue' not accessible: 'Bad file descriptor'."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-remoted","level":"error","description":"(1210): Queue '/queue/ossec/queue' not accessible: 'Bad file descriptor'."}
{"timestamp":"2019/01/21 11:45:21","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:22","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:22","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:22","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:23","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:24","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:25","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:26","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
{"timestamp":"2019/01/21 11:45:27","tag":"ossec-logcollector","level":"error","description":"socketerr (not available)."}
```

Other issues found:
```
(1210): Queue '/var/ossec/queue/ossec/queue' not accessible: 'Connection refused'.
(1211): Unable to access queue: '/var/ossec/queue/ossec/queue'. Giving up..
(1224): Error sending message to queue.
socketerr (not available).
(1211): Unable to access queue: '/var/ossec/queue/ossec/queue'. Giving up..
(1224): Error sending message to queue.
(1210): Queue '/queue/ossec/queue' not accessible: 'Connection refused'.
(1211): Unable to access queue: '/queue/ossec/queue'. Giving up..
```

Bug description
----------------
When upgrading from 3.7.2 to 3.8.0 on our Debian install, the following errors started arising in the ossec.json alerts file. logcollector would not take in any logs, and just kept spitting out the 'socketerr' error.

Looks to revolve around the queue

Had to revert back to 3.7.2 to get this functioning again.

**Comment by iasdeoupxe** [comment]
Might be related / unrelated to https://github.com/wazuh/wazuh/issues/2088. Have noticed [the same](https://github.com/wazuh/wazuh/issues/2088#issuecomment-455823094) on Debian after Upgrading and it seems (at least on my system) the ossec-remoted is crashing causing the missing queue.

**Comment by iasdeoupxe** [comment]
@Justangatang Any chances that you have a OpenLDAP system running where you're evaluating the logfile via an agent?

Have noticed https://github.com/wazuh/wazuh/issues/2453 today with a possible fix for this in https://github.com/wazuh/wazuh/pull/2456

**Comment by Justangatang** [comment]
@iasdeoupxe
We sure do have a system sending it's OpenLDAP log. I'll check to see if that system was registered around agent 33. 

Thank you for the update!

**Comment by vikman90** [comment]
Hi @Justangatang,

First off, sorry for the late response. Thank you for reporting this bug. Let me recap the related bugs:

- Analysisd 3.7.1 crashes when an agent 3.7.0 running on Windows reports an FIM scan (#2088). It was fixed on 3.7.2 (#2089).
- Analysisd 3.8.0 and 3.8.1 crashes when decoding OpenLDAP logs (#2453). It has just been fixed (#2456) on branch 3.8, we are releasing v3.8.2 including this patch on January 30.

Since removing OpenLDAP logs stopped the issue, I assume it's related to the latter. Sorry for the inconveniences.

Let me close this issue as we think it's solved. Please don't hesitate to reopen it if you have a further problem or question.

Thank you for your feedback.
Best regards.
